physics/discrete_trajectory2.hpp

```diff
@@ -0,0 +1,110 @@
+#pragma once
+
+#include <list>
+#include <memory>
+
+#include "absl/container/btree_map.h"
+#include "base/not_null.hpp"
+#include "geometry/named_quantities.hpp"
+#include "physics/degrees_of_freedom.hpp"
+#include "physics/discrete_trajectory_iterator.hpp"
+#include "physics/discrete_trajectory_segment_iterator.hpp"
+#include "physics/discrete_trajectory_types.hpp"
+#include "physics/trajectory.hpp"
+#include "serialization/physics.pb.h"
+
+namespace principia {
+namespace physics {
+
+template<typename Frame>
+class DiscreteTrajectorySegment;
+
+namespace internal_discrete_trajectory {
+
+using base::not_null;
+using geometry::Instant;
+using geometry::Position;
+using geometry::Velocity;
+using physics::DegreesOfFreedom;
+
+template<typename Frame>
+class DiscreteTrajectory2 : public Trajectory<Frame> {
+ public:
+  using Iterator = DiscreteTrajectoryIterator<Frame>;
```

Comment:
let’s call the result of `begin` `iterator`; the API is meant to look like a [_Container_](https://en.cppreference.com/w/cpp/named_req/Container) here (or a [`std::span`](https://en.cppreference.com/w/cpp/container/span) or somesuch).

Reply:
Done.